parser/kw.txt

```diff
@@ -0,0 +1,707 @@
+  | <CURRENT_USER: "CURRENT_USER">
+  | <CURSOR: "CURSOR">
+  | <CYCLE: "CYCLE">
+  | <DATE: "DATE"> 
+  | <DAY: "DAY">  // made it non reserved  as lot of people use it
+  | <DAYS: "DAYS">  // made it non reserved  as lot of people use it
+  | <DEC: "DEC">
+  | <DYNAMIC: "DYNAMIC">
+  | <EXP: "EXP">
+  | <EXTERNAL: "EXTERNAL">
+  | <FILTER: "FILTER">
+  | <FLOOR: "FLOOR">
+  | <FREE: "FREE">
+  | <FUNCTION: "FUNCTION">
+  | <GLOBAL: "GLOBAL">
+  | <HOLD: "HOLD">
+  | <HOUR: "HOUR">
+  | <HOURS: "HOURS">
+  | <IDENTITY: "IDENTITY">
+  | <INDICATOR: "INDICATOR">
+  | <INTERSECTION: "INTERSECTION">
+  | <INTERVAL: "INTERVAL">
+  | <LANGUAGE: "LANGUAGE">
+  | <LEAD: "LEAD">
+  | <LOCAL: "LOCAL">
+  | <LOWER: "LOWER">
+  | <MAX: "MAX">
+  | <MERGE: "MERGE">
+  | <METHOD: "METHOD">
+  | <MIN: "MIN">
+  | <MINUTE: "MINUTE">
+  | <MINUTES: "MINUTES">
+  | <MOD: "MOD">
+  | <MODULE: "MODULE">
+  | <MONTH: "MONTH">
+  | <MONTHS: "MONTHS">
+  | <NAME: "NAME">
+  | <NEW: "NEW">
+  | <NONE: "NONE">
+  | <OFFSET: "OFFSET">
+  | <OLD: "OLD">
+  | <OPEN: "OPEN">
+  | <PARTITION: "PARTITION">
+  | <POSITION: "POSITION">
+  | <POWER: "POWER">
+  | <PRECISION: "PRECISION">
+  | <RANGE: "RANGE">
+  | <RANK: "RANK">
+  | <REF: "REF">
+  | <REFERENCES: "REFERENCES">
+  | <RELEASE: "RELEASE">
+  | <RESULT: "RESULT">
+  | <RETURNS: "RETURNS">
+  | <ROLLUP: "ROLLUP">
+  | <ROW: "ROW">
+  | <ROW_NUMBER: "ROW_NUMBER">
+  | <ROWS: "ROWS">
+  | <SAVEPOINT: "SAVEPOINT">
+  | <SCOPE: "SCOPE">
+  | <SEARCH: "SEARCH">
+  | <SECOND: "SECOND">
+  | <SECONDS: "SECONDS">
+  | <SESSION_USER: "SESSION_USER">
+  | <SQL: "SQL">
+  | <START: "START">
+  | <STATIC: "STATIC">
+  | <SUM: "SUM">
+  | <SYSTEM: "SYSTEM">
+  | <TIME: "TIME">  // made it non reserved  as lot of people use it
+  | <TIMESTAMP: "TIMESTAMP">  // made it non reserved  as lot of people use it
+  | <TIMEZONE_HOUR: "TIMEZONE_HOUR">
+  | <TIMEZONE_MINUTE: "TIMEZONE_MINUTE">
+  | <TRIGGER: "TRIGGER">
+  | <TRUNCATE: "TRUNCATE">
+  | <UNKNOWN: "UNKNOWN">
+  | <UPDATE: "UPDATE">
+  | <UPPER: "UPPER">
+  | <USER: "USER">
+  | <VALUE: "VALUE">
+  | <VALUES: "VALUES">
+  | <VERSION: "VERSION">
+  | <VERSIONS: "VERSIONS">
+  | <WINDOW: "WINDOW">
+  | <YEAR: "YEAR">
+  | <YEARS: "YEARS">
+
+  // Presto tokens
+  | <COMMENT: "COMMENT">
+  | <DEFAULT_: "DEFAULT">
+  | <USE: "USE">
+  | <LIMIT: "LIMIT">
+  | "NUMERIC_HISTOGRAM"
+  | <REPLACE: "REPLACE">
+  | "HISTOGRAM"
+  | "APPROEX_PERCENTILE"
+  | "MAP_AGG"
+  | "SET_AGG"
+  | "MAP_UNION"
+  | <COUNT_QUOTED: "\"COUNT\"">
+}
+
+<UNREACHABLE>SKIP:
+{
+  // Dummy token to get a value range. Will never be mached:
+  <MAX_NON_RESERVED_WORD: "MAX NON RESERVED WORD">
+}
+
+<UNREACHABLE>SKIP:
+{
+  // Dummy token to get a value range. Will never be mached. And it should be here positionally - before the first non reserved word
+  <MIN_RESERVED_WORD: "MIN RESERVED WORD">
+}
+
+// reserved words
+TOKEN:
+{
+    //<ABS: "ABS">
```

Comment:
I think the ones commented are made non-reserved against the spec. Since these are already mentioned above, I think you can just remove them here. Or if you prefer to keep the original reserved list, maybe add a comment to explain this.

Reply:
Done